### PR TITLE
issue-3-edit-an-entry

### PR DESCRIPTION
**Pull request by alessandragutierrez**
code-journal-#3-edit-an-entry
code-journal-#3-mobile
code-journal-#3-desktop

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -18,83 +21,143 @@ function updatePhoto(event) {
...
-  } else $entriesEmpty.classList.remove('hidden');
+  data.nextEntryId++;
+  data.entries.unshift(newEntry);
+}
 
-  swapViews(data.view);
-});
+function addEntryToPage() {
+  i = 0;
+  var entryValues = renderEntry(data.entries[i]);
```
Comment (codingcodymiller) [inline js/main.js]: I think you might be able to do away with this global `i` variable. In both cases I see you using it, you are assigning a value to it on the line above. Instead, you can just use that value directly. for instance here, you can just pass `data.entries[0]` as the argument to `renderEntry`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Great work!! One note for you:

**Comment by alessandragutierrez** [comment]
code-journal-#3-edit-entry-title

File: js/main.js [hunk 2]
```diff
@@ -8,6 +8,10 @@ var $notes = document.querySelector('.notes');
 var $ul = document.querySelector('ul');
 var $viewElements = document.querySelectorAll('.view');
 var $entriesEmpty = document.querySelector('.entries-empty');
+var $pageTitle = document.querySelector('.page-title');
+var dataEntryId;
+var entryToEdit;
+var editedEntryIndex;
 var i;
```
Comment (codingcodymiller) [inline js/main.js]: You should be able to remove this now.

File: js/main.js [hunk 3]
```diff
@@ -18,83 +22,152 @@ function updatePhoto(event) {
...
+    notes: $notes.value
+  };
+
+  data.entries.splice(editedEntryIndex, 1, editedEntry);
+  data.editing = null;
+}
+
+function replaceEntryOnPage() {
+  i = editedEntryIndex;
+  var entryValues = renderEntry(data.entries[i]);
```
Comment (codingcodymiller) [inline js/main.js]: This should get adjusted too, just like the other use of `i`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**